presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -203,10 +205,15 @@ private Node rewriteShowStats(ShowStats node, Table table, Constraint<ColumnHand
                         .filter(entry -> columns.contains(entry.getKey()))
                         .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
             }
-            TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, ImmutableList.copyOf(columnHandles.values()), constraint);
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+            List<ColumnHandle> nonHiddenColumns = ImmutableList.copyOf(tableMetadata.getColumns().stream().filter(column -> !column.isHidden())
+                    .map(ColumnMetadata::getName)
+                    .map(columnHandles::get)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()));
+            TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, nonHiddenColumns, constraint);
```

Comment:
I see that this part is originally from https://github.com/prestodb/presto/pull/22327
Can you explain what this change is and how does it affect the non-iceberg tables?

Reply:
This prevents hidden columns from being queried when a user executes `SHOW STATS`. The `HiveMetadata` already implicitly filters out hidden columns when returning from `getTableStatistics` without a constraint.